src/wallet/rpcdump.cpp

```diff
@@ -832,13 +837,18 @@ UniValue z_importviewingkey(const UniValue& params, bool fHelp)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid viewing key");
     }
 
+    auto addrInfo = boost::apply_visitor(libzcash::AddressInfoFromViewingKey{}, viewingkey);
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("type", addrInfo.first);
+    result.pushKV("address", EncodePaymentAddress(addrInfo.second));
+
     auto addResult = boost::apply_visitor(AddViewingKeyToWallet(pwalletMain), viewingkey);
     if (addResult == SpendingKeyExists) {
         throw JSONRPCError(
             RPC_WALLET_ERROR,
             "The wallet already contains the private key for this viewing key");
     } else if (addResult == KeyAlreadyExists && fIgnoreExistingKey) {
-        return NullUniValue;
+        return result;
     }
     pwalletMain->MarkDirty();
```

Comment:
This marks the wallet dirty even if no change was made (i.e. the result was not `KeyAdded`) and there will be no rescan. That's harmless but unnecessary. I would have suggested a change here, but it's not at all clear why the responsibility for marking the wallet dirty should lie with this code and not `wallet.cpp`. Perhaps open a ticket.

Reply:
https://github.com/zcash/zcash/issues/4428